service/history/workflow/mutable_state_impl.go

```diff
@@ -136,6 +136,8 @@ var (
 	timeZeroUTC = time.Unix(0, 0).UTC()
 )
 
+var emptyTasks = []tasks.Task{}
```

Comment:
I would remove this thing completely and use `nil` everywhere. I understand that it is not part of this PR.